src/encoding.rs

```diff
@@ -111,9 +103,20 @@ impl Decoder {
     ///
     /// If XML declaration is absent in the XML, UTF-8 is used.
     ///
+    /// ----
     /// Returns an error in case of malformed sequences in the `bytes`.
     pub fn decode_with_bom_removal<'b>(&self, bytes: &'b [u8]) -> Result<Cow<'b, str>> {
```

Comment:
I'm not sure, should we keep this method? It is not used internally, maybe remove it?

Reply:
Delete it

Reply:
The free function `decode_with_bom_removal` also is not used currently. Do you plan to use it in one of you PRs? Otherwise I suggest to remove it also. It seems that in any case, the user will not have to take care of the BOM, it will be handled by quick-xml internally and new access methods that could be introduced in #461 probably also will not require that function

Reply:
I don't, we can remove it.  As you say, the options provided to `Reader` ought to be able to handle this eventually.